src/pkg/user/service/service.go

```diff
@@ -22,77 +28,247 @@ type Service struct {
+	_, userPrivateKey, err := s.Infra.Keystore.Get(userID)
+	if err != nil {
+		return eth_common.Address{}, nil, fmt.Errorf("Keystore.Get error: %w userID %s", err, userID)
 	}
 
 	privateUserKey := userPrivateKey[:]
 
 	privateKey, err := crypto.ToECDSA(privateUserKey)
 	if err != nil {
-		return fmt.Errorf("crypto.ToECDSA error: %w userID %s", err, user.UserID)
+		return eth_common.Address{}, nil, fmt.Errorf("crypto.ToECDSA error: %w userID %s", err, userID)
 	}
 
 	address := crypto.PubkeyToAddress(privateKey.PublicKey)
-	ehrSystemID := ctx.(*gin.Context).GetString("ehrSystemID")
 
-	pwdHash, err := s.generateHash(user.UserID, ehrSystemID, user.Password)
+	pwdHash, err := s.generateHash(privateUserKey, userID, ehrSystemID, password)
 	if err != nil {
-		return fmt.Errorf("register s.generateHash error: %w userID %s, password: %s", err, user.UserID, user.Password)
+		return eth_common.Address{}, nil, fmt.Errorf("register s.generateHash error: %w userID %s, password: %s", err, userID, password)
 	}
+	return address, pwdHash, nil
+}
 
-	requestID := ctx.(*gin.Context).GetString("reqId")
+// method should be idempotent
+func (s *Service) generateHash(salt []byte, phrases ...string) ([]byte, error) {
+	hash := strings.Join(phrases, "")
 
-	txHash, err := s.Infra.Index.UserAdd(requestID, address, user.UserID, user.Role, pwdHash)
+	pwdHash, err := scrypt.Key([]byte(hash), salt, N, r, p, keyLen)
 	if err != nil {
-		return fmt.Errorf("Index.DeleteDoc error: %w", err)
+		return nil, fmt.Errorf("scrypt.Key error: %w", err)
 	}
 
-	procRequest.AddEthereumTx(proc.TxDeleteDoc, txHash)
+	return pwdHash, nil
+}
+
+func (s *Service) CreateToken(userID string) (*TokenDetails, error) {
+	td := &TokenDetails{}
+	td.AtExpires = time.Now().Add(common.JWTExpires).Unix()
+	td.AccessUUID = uuid.New().String()
+
+	td.RtExpires = time.Now().Add(common.JWTRefreshExpires).Unix()
+	td.RefreshUUID = td.AccessUUID + "++" + userID
+
+	var err error
+	//Creating Access Token
+	accessTokenSecret, _, err := s.Infra.Keystore.Get(userID)
+	if err != nil {
+		return nil, fmt.Errorf("CreateToken Keystore.Get error: %w userID %s", err, userID)
+	}
 
+	refreshTokenSecret, _, err := s.Infra.Keystore.Get(userID + "_refresh")
+	if err != nil {
+		return nil, fmt.Errorf("CreateRefreshToken Keystore.Get error: %w userID %s", err, userID)
+	}
+
+	atClaims := jwt.MapClaims{}
+	atClaims["uuid"] = td.AccessUUID
+	atClaims["user_id"] = userID
+	atClaims["exp"] = td.AtExpires
+	// TODO to fill user metadata like roles we should create new method in contract i.e. UserGet!!!
+	at := jwt.NewWithClaims(jwt.SigningMethodHS256, atClaims)
+	td.AccessToken, err = at.SignedString((*accessTokenSecret)[:])
+
+	if err != nil {
+		return nil, err
+	}
+
+	//Creating Refresh token
+	rtClaims := jwt.MapClaims{}
+	rtClaims["uuid"] = td.RefreshUUID
+	rtClaims["user_id"] = userID
+	rtClaims["exp"] = td.RtExpires
+	rt := jwt.NewWithClaims(jwt.SigningMethodHS256, rtClaims)
```

Comment:
Use ECDSA signing method as written in the task. Use the user's private key from the keystore as the key